modules/core/src/main/java/org/locationtech/jts/io/WKTReader.java

```diff
@@ -222,170 +437,255 @@ public Geometry read(Reader reader) throws ParseException {
+  }
+
+  /**
+   * Returns the next array of <code>Coordinate</code>s in the stream.
+   *
+   *@param  tokenizer        tokenizer over a stream of text in Well-known Text
+   *      format. The next element returned by the stream should be a number.
+   *@return                  the next array of <code>Coordinate</code>s in the
+   *      stream.
+   *@throws  IOException     if an I/O error occurs
+   *@throws  ParseException  if an unexpected token was encountered
+   *
+   *@deprecated in favor of functions returning {@link CoordinateSequence}s
+   */
+  private Coordinate[] getCoordinatesNoLeftParen(StreamTokenizer tokenizer) throws IOException, ParseException {
+    String nextToken = null;
+    ArrayList coordinates = new ArrayList();
+    coordinates.add(getPreciseCoordinate(tokenizer));
+    nextToken = getNextCloserOrComma(tokenizer);
+    while (nextToken.equals(COMMA)) {
+      coordinates.add(getPreciseCoordinate(tokenizer));
+      nextToken = getNextCloserOrComma(tokenizer);
+    }
+    Coordinate[] array = new Coordinate[coordinates.size()];
+    return (Coordinate[]) coordinates.toArray(array);
+  }
+
+  /**
+   * Returns the next precise <code>Coordinate</code> in the stream.
+   *
+   *@param  tokenizer        tokenizer over a stream of text in Well-known Text
+   *      format. The next element returned by the stream should be a number.
+   *@return                  the next array of <code>Coordinate</code>s in the
+   *      stream.
+   *@throws  IOException     if an I/O error occurs
+   *@throws  ParseException  if an unexpected token was encountered
+   *
+   *@deprecated in favor of functions returning {@link CoordinateSequence}s
    */
-  private Coordinate[] getCoordinates() throws IOException, ParseException {
-		String nextToken = getNextEmptyOrOpener();
-		if (nextToken.equals(EMPTY)) {
-			return new Coordinate[] {};
-		}
-		ArrayList coordinates = new ArrayList();
-		coordinates.add(getPreciseCoordinate());
-		nextToken = getNextCloserOrComma();
-		while (nextToken.equals(COMMA)) {
-			coordinates.add(getPreciseCoordinate());
-			nextToken = getNextCloserOrComma();
-		}
-		Coordinate[] array = new Coordinate[coordinates.size()];
-		return (Coordinate[]) coordinates.toArray(array);
-	}
-
-	private Coordinate[] getCoordinatesNoLeftParen() throws IOException, ParseException {
-		String nextToken = null;
-		ArrayList coordinates = new ArrayList();
-		coordinates.add(getPreciseCoordinate());
-		nextToken = getNextCloserOrComma();
-		while (nextToken.equals(COMMA)) {
-			coordinates.add(getPreciseCoordinate());
-			nextToken = getNextCloserOrComma();
-		}
-		Coordinate[] array = new Coordinate[coordinates.size()];
-		return (Coordinate[]) coordinates.toArray(array);
-	}
-
-  private Coordinate getPreciseCoordinate()
+  private Coordinate getPreciseCoordinate(StreamTokenizer tokenizer)
       throws IOException, ParseException
   {
     Coordinate coord = new Coordinate();
```

Comment:
Should this method consider making different kinds of Coordinates? The CoordinateSequence implementations have a createCoordainte() method that may help, also the Coordinates utility class.